@here/harp-geoutils/lib/coordinates/GeoPolygon.ts

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright (C) 2017-2020 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { Vector2 } from "three";
+
+import { GeoBox } from "./GeoBox";
+import { GeoCoordinates, MAX_LATITUDE, MIN_LATITUDE } from "./GeoCoordinates";
+import { GeoCoordinatesLike } from "./GeoCoordinatesLike";
+import { GeoCoordLike, geoCoordLikeToGeoCoordinatesLike } from "./GeoCoordLike";
+import { GeoPolygonLike } from "./GeoPolygonLike";
+
+type MinThreeItemsArray<T> = [T, T, T, ...T[]];
+
+export type GeoPolygonCoordinates = MinThreeItemsArray<
+    GeoCoordinatesLike | GeoCoordinates | GeoCoordLike
+>;
+
+/**
+ * A GeoPolygon in 2D Space (altitudes will be ignored).
+ * Coordinates are expected in counter-clockwise order, for concave polygons a sorting is
+ * available.
+ * Clockwise ordered or selfintersecting Polygons might lead to no or unexpected results.
+ *
+ * @beta @internal
+ */
+export class GeoPolygon implements GeoPolygonLike {
+    coordinates: GeoPolygonCoordinates;
```

Comment:
Shouldn't this be private and we should only have a readable property or is it a valid usecase to add points directly to the coordinates array? Especially when they were sorted before by the ctor it would be weird if users would add them randomly.

Reply:
done